source/Plugins.cpp

```diff
@@ -0,0 +1,124 @@
+/* Plugins.cpp
+Copyright (c) 2022 by Sam Gleske (samrocketman on GitHub)
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Plugins.h"
+
+#include "DataFile.h"
+#include "DataNode.h"
+#include "DataWriter.h"
+#include "Files.h"
+
+#include <algorithm>
+#include <map>
+
+using namespace std;
+
+namespace {
+	map<string, bool> settings;
+	map<string, bool> frozenSettings;
+
+	void LoadSettings(const string &path)
+	{
+		DataFile prefs(path);
+		for(const DataNode &node : prefs)
+		{
+			const string &key = node.Token(0);
+			if(key == "plugins")
+				for(const DataNode &child : node)
+					if(child.Size() == 2)
+						settings[child.Token(0)] = child.Value(1);
+		}
+	}
+}
+
+
+
+void Plugins::Load()
+{
+	// Global plugin settings
+	LoadSettings(Files::Resources() + "plugins.txt");
```

Comment:
Do we need global plugin settings? One plugin preference file aught to be enough.

Reply:
Yes we do.  I already use it.  We have global plugins so supporting a global settings definition makes sense.

If you'd like an example see https://github.com/samrocketman/endless-sky/blob/mining-complete-plugins/plugins.txt

These are set upon first launch by a new player.  I think we should also allow plugins to define themselves as being disabled by default but I'll open a separate PR for that.

Reply:
This would also enable the following:

- Breaking out core game data from interface data which would allow players to disable core data for complete conversion mods.
- High DPI can be pre-packaged and disabled by default.  One of the original criticisms of prepackaging high DPI was its memory and load time requirements.
- Additional logic can be added to the game where if the resolution is above a certain point that High DPI is automatically enabled.